cpp/include/raft/neighbors/ivf_list.hpp

```diff
@@ -135,7 +136,7 @@ void serialize_list(const raft::device_resources& handle,
 template <template <typename> typename SpecT, typename IdxT, typename SizeT>
 void serialize_list(const raft::device_resources& handle,
                     std::ostream& os,
-                    const std::shared_ptr<const list<SpecT, IdxT, SizeT>>& ld,
+                    const std::shared_ptr<list<SpecT, IdxT, SizeT>>& ld,
```

Comment:
Is there a specific reason you removed the const modifier here, or is it just an artifact of reverting the previous changes? If the latter, please revert the changes in this file (const modifier is a part of the contract that says the lists are the inputs).

Reply:
Yes, I removed the const modifier here because it is breaking the template type inference and it is not what the function is being called with https://github.com/rapidsai/raft/blob/e7f0268c9f5c17756c415ed47e1856030b3f235d/cpp/include/raft/neighbors/ivf_pq_types.hpp#L497
I'm using this fix because passing `list_spec_wrapper<T>::list_spec` as the `SpecT` template parameter is leading to the following compilation error ```error: argument list for class template "raft::neighbors::ivf_flat::list_spec_wrapper<ValueT>::list_spec" is missing``` even though IVF-PQ seems to have no problem using `list_spec` without the template argument...

Reply:
You're right, I see it's more troublesome to modify `index::lists()` and other calls compared to having a little compromise like this.
I've played a little bit with the templates and suggest to change the signature to make it more flexible/convenient for both ivf-flat and ivf-pq. Here it is https://github.com/lowener/raft/pull/1